src_kt/java/arcs/storage/Driver.kt

```diff
@@ -0,0 +1,52 @@
+/*
+ * Copyright 2019 Google LLC.
+ *
+ * This code may only be used under the BSD style license found at
+ * http://polymer.github.io/LICENSE.txt
+ *
+ * Code distributed by Google as part of this project is also subject to an additional IP rights
+ * grant found at
+ * http://polymer.github.io/PATENTS.txt
+ */
+
+package arcs.storage
+
+/**
+ * Interface that all drivers must support.
+ *
+ * Note the threading of a version number here; each model provided by the [Driver] to the [Store]
+ * (using a receiver registered with [registerReceiver]) is paired with a version, as is each model
+ * sent from the [Store] to the driver (using [send]).
+ *
+ * This threading is used to track whether driver state has changed while the [Store] is processing
+ * a particular model. [send] should always fail if the version isn't exactly `1` greater than the
+ * current internal version.
+ */
+interface Driver<Data : Any> {
+  /** Key identifying the [Driver]. */
+  val storageKey: StorageKey
+
+  /** Requirement for the existence of the [Driver]. */
+  val existenceCriteria: ExistenceCriteria
```

Comment:
We should keep type & enum names in sync as much as possible; that doesn't mean we can't change the .ts variant but we should make sure any proposed changes are socialized & captured somewhere.

Reply:
Ok, sounds good. I thought `Existence` was too terse as I read through the code the first time.